feat(sh): improve install.sh to install in path

quasiuslikecautious opened the pull request:
Add --install-path to install.sh to allow user to specify where ockam binary should be installed. Additionally, set path in rcfiles following the implementation of rustup. Additionally add --version to specify version, and --no-modify-path to allow user to avoid modifying their path

<!-- Thank you for sending a pull request :heart: -->

## Current behavior
Running install.sh downloads and installs the ockam binary for use in the current directory, and the user must manually add the binary to their PATH.

<!-- Please describe the current behavior of the code before the changes in this pull request is applied. -->

## Proposed changes
As proposed in issue #4359 the user can now specify the install path of the binary with the --install-path PATH flag, and unless the --no-modify-path flag is set, the ockam binary will be added to the user's path. 

<!-- Please describe the changes proposed in this pull request. -->
<!-- If this pull request resolves an already recorded bug or a feature request, please add a link to that issue. -->

## Checks

<!-- To help us review and merge this pull request quickly, please confirm the following:  -->

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will mark the commit as verified.

<!-- Looking forward to merging your contribution!! -->

quasiuslikecautious commented:
Hey there, this is my first PR is an Open Source Project! Would love some pointers/am happy to provide any additional info or changes!

mrinalwadhwa commented:
@quasiuslikecautious Awesome, thank you for spending time on this!

I'll take some time to read soon.

In the meantime, before we can merge please

- [x] Accept the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding your Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will make the commit as verified.

mrinalwadhwa commented on install.sh:
```diff
@@ -155,38 +166,182 @@ download() {
+    add_to_path
+  fi
+
+  echo "$_modify_path"
```
This line could be removed.

quasiuslikecautious replied:
Ah - must have missed that when removing debug lines. Line has been removed in latest changes!

mrinalwadhwa commented:
@quasiuslikecautious the script works great!

A couple of small changes would make it even better:

1. Could you resolve the shellcheck lints here
https://github.com/build-trust/ockam/actions/runs/4443751111/jobs/7802157817?pr=4480

2. Run the script through [shfmt](https://github.com/sambacha/shfmt)

3. Another thing I noticed is that rustup writes:

```
. "$HOME/.cargo/env"
```

however the install.sh from the PR writes:

```
. "/Users/mrinal/.ockam/env"
```

maintaining the `$HOME` in the rcfiles might be less brittle across shells.

quasiuslikecautious commented:
Sounds good, and thanks for the feedback! I removed the unnecessary echo line, resolved shellcheck lint error, ran and formatted code with shfmt, and added functionality to replace the expanded $HOME path with the '$HOME' variable in the env script and the rcfile appended source command.

As a note, I appended the commit and force pushed to the branch to keep everything within one commit. Let me know if there's any other improvements to add/changes to make!

mrinalwadhwa commented on install.sh:
```diff
@@ -93,6 +93,35 @@ required() {
+# eval to expand user inputted filepath
+expand_filepath() {
+  local _path="$1"
+  _path="${_path/#\./$(pwd)}"
```
@quasiuslikecautious This is looking pretty close to ready except for on minor issue.
shfmt fails on this line because we're expecting any posix shell and search and replace is a bash/mksh feature
https://github.com/build-trust/ockam/actions/runs/4448885465/jobs/7812303056?pr=4480

P.S I force pushed to your branch to tackle some minor lints, mainly that we use spaces for indenting scripts.

quasiuslikecautious replied:
Just pushed latest changes in new commit (and also resolved merge conflict)  - switched expand_filepath to use sed + regex to replace filepath symbols instead of bash command. Let me know if any other changes are needed, and thanks again for the feedback/reviewing!

mrinalwadhwa reviewed (changes requested):
This looks close to merge after we tackle the comment above in expand_filepath

quasiuslikecautious committed:
build: improve install.sh to install in path

* Change to only run with bash
* Add --install-path to specify where ockam binary should be installed.
* Create an env file that will set path in detected rcfiles.
* Add --version to specify version
* Add --no-modify-path to allow user to avoid modifying their path

mrinalwadhwa committed:
docs: update install instructions in readme

mrinalwadhwa commented:
@quasiuslikecautious Congratulations on an awesome first contribution 🥳 🎉
